rust/agama-lib/share/profile.schema.json

```diff
@@ -568,26 +560,227 @@
+      "pattern": "^[0-9]+(\\.[0-9]+)?(\\s*([KkMmGgTtPpEeZzYy][iI]?)?[Bb])?$",
+      "examples": ["2 GiB", "3.5 MiB"]
     },
     "sizeInteger": {
       "title": "Size in bytes",
       "type": "integer",
-      "minimum": 0
+      "minimum": 0,
+      "examples": [1024, 2048]
     },
     "sizeValue": {
       "anyOf": [
         { "$ref": "#/$defs/sizeString" },
         { "$ref": "#/$defs/sizeInteger" }
       ]
+    },
+    "size": {
+      "title": "Size options.",
+      "anyOf": [
+        {
+          "title": "Automatic size.",
+          "description": "The size is auto calculated according to the product.",
+          "const": "auto"
+        },
+        {
+          "title": "Size unit.",
+          "$ref": "#/$defs/sizeValue"
+        },
+        {
+          "title": "Size range.",
+          "description": "Lower size limit and optionally upper size limit.",
+          "type": "array",
+          "items": {
+            "$ref": "#/$defs/sizeValue"
+          },
+          "minItems": 1,
+          "maxItems": 2,
+          "examples": [[1024, 2048], ["1 GiB", "5 GiB"], [1024, "2 GiB"], ["2 GiB"]]
+        },
+        {
+          "title": "Size range.",
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["min"],
+          "properties": {
+            "min": {
+              "title": "Mandatory lower size limit.",
+              "$ref": "#/$defs/sizeValue"
+            },
+            "max": {
+              "title": "Optional upper size limit.",
+              "$ref": "#/$defs/sizeValue"
+            }
+          }
+        }
+      ]
+    },
+    "search": {
+      "title": "Search options",
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["name"],
+      "properties": {
+        "name": {
+          "title": "Device name",
+          "type": "string",
+          "examples": ["/dev/vda"]
+        }
+      }
+    },
+    "boot": {
+      "title": "Boot options.",
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["configure"],
+      "properties": {
+        "configure": {
+          "title": "Whether to configure partitions for booting.",
+          "type": "boolean"
+        },
+        "device": {
+          "title": "Device to use for booting.",
+          "description": "The installation device is used by default for booting.",
+          "type": "string",
+          "examples": ["/dev/vda"]
+        }
+      }
+    },
+    "encryptionMethod": {
+      "title": "Method used to create the encryption device.",
+      "enum": ["luks2", "tpm_fde"]
+    },
+    "encryptionPbkdFunction": {
+      "title": "Password-based key derivation function to use for LUKS2.",
+      "enum": ["pbkdf2", "argon2i", "argon2id"]
+    },
+    "encryption": {
+      "title": "Encryption options.",
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["key"],
+      "properties": {
+        "key": {
```

Comment:
I don't like the name "key" because is too ambiguous (not to say plain wrong). With LUKS there is a _master key_, several _user keys_,  potentially _FIDO2 security keys_ and more types. And guess what, none of them correspond to the passphrase. The passphrase is not a "key" (the actual user keys are derived from the passphrase using the PBKDF).

So the correct name here would be `password` or `passphrase`.

Reply:
Ok. I have renamed it as `password`.